### DISCUSSION TITLE
STDIN and processing all files in sub directories

### DISCUSSION THREAD

**Discussion by valdier1** [ImageMagick/ImageMagick#5044] [Help]
I am running into a bit of a syntax issue here and I am sure someone here has tackled it previously.

I am trying to resize 10's of thousands of images in fairly deep directory structure

I am trying to do similar to this:

`dir C:\testfolder\*.001 /b/s | magick mogrify -resize 1500> -format tif `

This obviously isn't the right syntax but I'm not sure why it isn't. The tif files have an extension of .001, this should pipe all of them and it works if I run the second command as:

`magick mogrify *.001 -resize 1500> -format tif `

But it obviously doesn't account for subdirectories... any ideas?

**Comment by fmw42** [comment]
For all image:

`magick mogrify -resize 1500> -format tif *`

or for just tif images

`magick mogrify -resize 1500> -format tif *.tif`

**Reply by fmw42** [reply]
You will have to write a script FOR loop over your subdirectories and call magick mogrify for each subdirectory.

**Reply by valdier1** [reply]
hmmm that might be what I'm having the problem with, is feeding those values to magick

Also 
`magick mogrify -resize 1500> -format tif *.*`
doesn't work, it has to be 
`magick mogrify *.* -resize 1500> -format tif`

**Reply by valdier1** [reply]
This should work: `FOR /R %%a IN (*.*) DO magick mogrify "%%~a" -resize 500^> -format tif`

but it doesn't for some reason... with or without the > escape character

**Reply by valdier1** [reply]
`FOR /R %%a IN (*.*) DO magick mogrify %%a -resize 500^> -format tif`

Seems to be the winning combo to get it working

**Comment by valdier1** [comment]
Ok, I got the script working, but its not actually resizing:

`magick mogrify c:\testfolder\D1928184.001 -resize 500> -format tif`

shows that it completes without errors, but doesn't actually resize the image... what am I missing here? :)  It starts off as a 2000x3000 image and after running, its still 2000x3000

**Reply by fmw42** [reply]
You over looked my first answer about ending with * or *.tif

`magick mogrify -resize 500> -format tif *`

for all images

`magick mogrify -resize 500> -format tif *.tif`

for just tif images

The * or *.tif tells mogrify which images to use as input. The -format tif specifies the output format.

**Reply by valdier1** [reply]
If I add the * at the end I get:

`
C:\testfolder\D1928184.001 -resize 500> -format tif *
mogrify: unable to open image 'tif': No such file or directory @ error/blob.c/OpenBlob/3527.
mogrify: no decode delegate for this image format `' @ error/constitute.c/ReadImage/737.
mogrify: unable to open image '*': Invalid argument @ error/blob.c/OpenBlob/3527.
mogrify: unable to open image '*': Invalid argument @ error/blob.c/OpenBlob/3527.
mogrify: no decode delegate for this image format `' @ error/constitute.c/ReadImage/737.
mogrify: no decode delegate for this image format `' @ error/constitute.c/ReadImage/737.
mogrify: unable to open module file 'C:\Program Files\ImageMagick-7.1.0-Q16-HDRI\modules\coders\IM_MOD_RL_BAT_.dll': No such file or directory @ warning/module.c/GetMagickModulePath/675.
mogrify: no decode delegate for this image format `BAT' @ error/constitute.c/ReadImage/737
.`

**Reply by fmw42** [reply]
What was your command? It should just be:

magick mogrify -resize 500> -format tif *

Does that work?

**Reply by fmw42** [reply]
What format is .001?  That is not an image format?  What is the proper suffix type for your input file?

Or try

`magick mogrify -resize 500> -format tif D1928184.001`

The input goes at the end or use * or *.suffix

**Reply by valdier1** [reply]
It didn't, but I think I figured out why now... 

`magick mogrify -resize "500>" -format tif *`

it needed the quotes around the >

but it renamed it to D1928184.tif

**Reply by fmw42** [reply]
Yes on some systems you either need double quotes or your need to escape your > with your OS's escape character. But double quotes is good.

**Reply by valdier1** [reply]
Ok, so last issue

`magick mogrify -resize "1000>" -depth 4 -compress LZW -format tif *`

creates a new image with the .tif extension rather than doing an in place change to the original file

**Reply by fmw42** [reply]
Because your input is .001 and not the proper .suffix and the output is a different format as .tif

**Reply by fmw42** [reply]
You told mogrify to use tif for the output when you use -format tif.  That is the output, not an input selector. If you want the same image overwritten, then specify the exact input file with suffix at the end in place of * and leave off the -format tif

**Reply by valdier1** [reply]
I totally understand, thank you a lot. I've got it working now and I 100% appreciate all of the help you have given on this
